src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+                Tuple<string, List<OpenApiParameter>> pkComponents = CreatePrimaryKeyPathComponentAndParameters(entityName);
+                string pkPathComponents = pkComponents.Item1;
+                string fullPathComponent = entityBasePathComponent + pkPathComponents;
+
+                OpenApiOperation getOperation = new()
+                {
+                    Description = GETONE_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                };
+                getOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+
+                OpenApiOperation putOperation = new()
+                {
+                    Description = PUT_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
+                };
+                putOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+                putOperation.Responses.Add(HttpStatusCode.Created.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Created), responseObjectSchemaName: entityName));
+
+                OpenApiOperation patchOperation = new()
+                {
+                    Description = PATCH_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
+                };
+                patchOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+                patchOperation.Responses.Add(HttpStatusCode.Created.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Created), responseObjectSchemaName: entityName));
+
+                OpenApiOperation deleteOperation = new()
+                {
+                    Description = DELETE_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
+                };
+                deleteOperation.Responses.Add(HttpStatusCode.NoContent.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.NoContent)));
+
+                OpenApiPathItem openApiPathItem = new()
+                {
+                    Operations = new Dictionary<OperationType, OpenApiOperation>()
+                    {
+                        // Creation GET, POST, PUT, PATCH, DELETE operations
+                        [OperationType.Get] = getOperation,
```

Comment:
I remember for a stored proc entity the default is POST operation.
Should we handle stored procs separately - create only those operation that are specified with overridden behavior in the config file?

Reply:
good point, that would probably be best since I'm adding both currently. I also am not properly handling SP parameters, so i will add that fix too.

Reply:
updated code to now how separate code path for generating stored procedure schema component definitions. Stored procedures require parameters to be set in the request body (except GET , where params go in query string), so a request body will always include the Params for the stored procs.